### DISCUSSION TITLE
Why `measure` on Android doesn't calculate relative position?

### DISCUSSION THREAD

**Discussion by janlat** [software-mansion/react-native-reanimated#3730] [Q&A]
`measure` on Android is hardcoded to return `0` for both x and y. Are there any technical reasons why this wasn't implemented?

**Comment by tomekzaw** [comment]
How do you know it's hardcoded to return 0? As far as I know it's not.

**Reply by janlat** [reply]
It always returns 0 for me, so I tried debugging a bit and found this: 
https://github.com/software-mansion/react-native-reanimated/blob/8933a00fce0fa2b3f636c83f75ad708bfbb3e9d0/android/src/main/java/com/swmansion/reanimated/NativeMethodsHelper.java#L32-L34
but maybe I'm missing something here

**Reply by tomekzaw** [reply]
Well, didn't notice this before 😅 I will investigate this further and I will let you know.

**Reply by tomekzaw** [reply]
I'm back with the results.

Example:

<img width="250" alt="screenshot" src="https://user-images.githubusercontent.com/20516055/199772136-c45ff913-83ce-4ffd-8207-fcd4e8b6d5ea.png">

Library:
* React Native - `ref.current.measure(...)`, https://reactnative.dev/docs/direct-manipulation#measurecallback
* Reanimated - `measure(animatedRef)`, https://docs.swmansion.com/react-native-reanimated/docs/api/nativeMethods/measure/

| Platform | Architecture | Library | Result |
|:-:|:-:|:-:|-|
| Android | Paper | React Native | `{"height": 80, "pageX": 20, "pageY": 256, "width": 80, "x": 0, "y": 0}` |
| Android | Paper | Reanimated | `{"height": 80, "pageX": 20, "pageY": 256, "width": 80, "x": 0, "y": 0}` |
| Android | Fabric | React Native | `{"height": 80, "pageX": 20, "pageY": 256, "width": 80, "x": 20, "y": 30.18181800842285}` |
| Android | Fabric | Reanimated | `{"height": 80, "pageX": 20, "pageY": 256, "width": 80, "x": 20, "y": 30.18181800842285}` |
| iOS | Paper | React Native | `{"height": 80, "pageX": 20, "pageY": 263, "width": 80, "x": 20, "y": 30}` |
| iOS | Paper | Reanimated | `{"height": 80, "pageX": 20, "pageY": 263, "width": 80, "x": 20, "y": 30}` |
| iOS | Fabric | React Native | `{"height": 80, "pageX": 20, "pageY": 263, "width": 80, "x": 20, "y": 30}` |
| iOS | Fabric | Reanimated | `{"height": 80, "pageX": 20, "pageY": 263, "width": 80, "x": 20, "y": 30}` |

It looks like in each case Reanimated matches the output from React Native, which is the expected behavior.

**Reply by janlat** [reply]
Huh, interesting. Another reason to migrate to the new architecture I guess :smile: 
Do you think it's worth mentioning in the docs?

**Reply by tomekzaw** [reply]
> Huh, interesting. Another reason to migrate to the new architecture I guess 😄

Maybe it's worth submitting an issue in [react-native](https://github.com/facebook/react-native) repo? Technically, the behaviour should be consistent across the architectures.

> Do you think it's worth mentioning in the docs?

Since the behavior is identical to the original from React Native, I don't think it's urgent. But feel free to open a PR if you find this piece of information useful.
